lib/arrow/disruptions/disruption_v2.ex

```diff
@@ -106,4 +107,12 @@ defmodule Arrow.Disruptions.DisruptionV2 do
       changeset
     end
   end
+
+  defp validate_no_mode_change(changeset) do
+    if !is_nil(changeset.data.mode) and get_field(changeset, :mode) != changeset.data.mode do
```

Comment:
suggestion: How about this?
```suggestion
    if Changeset.changed?(changeset, :mode) and not Changeset.changed?(changeset, :mode, from: nil) do
```